apps/comps/rpc/router/competitions/get-by-id.ts

```diff
@@ -0,0 +1,28 @@
+import { z } from "zod/v4";
+
+import { base } from "@/rpc/context/base";
+import { userMiddleware } from "@/rpc/middleware/user";
+
+export const getById = base
+  .use(userMiddleware)
```

Comment:
Using the user middleware here. This doesn't mean they are authenticated (there may not be a user), but the user is resolved, if possible, and it is undefined otherwise. This is because the service function to get a competition by id also accepts an optional user id, and will "enrich" the competition with user-specific data. I hate this, but that is the way it is.